src/pkg/services/m365/api/contacts_pager_test.go

```diff
@@ -0,0 +1,45 @@
+package api_test
+
+import (
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type ContactsPagerIntgSuite struct {
+	tester.Suite
+	cts clientTesterSetup
+}
+
+func TestContactsPagerIntgSuite(t *testing.T) {
+	suite.Run(t, &ContactsPagerIntgSuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tester.M365AcctCredEnvs}),
+	})
+}
+
+func (suite *ContactsPagerIntgSuite) SetupSuite() {
+	suite.cts = newClientTesterSetup(suite.T())
+}
+
+func (suite *ContactsPagerIntgSuite) TestGetItemsInContainerByCollisionKey() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	results, err := suite.cts.ac.Contacts().GetItemsInContainerByCollisionKey(ctx, suite.cts.userID, "contacts")
+	require.NoError(t, err, clues.ToCore(err))
+	require.Less(t, 0, len(results), "requires at least one result")
+
+	for k, v := range results {
+		assert.NotEmpty(t, k, "all keys should be populated")
+		assert.NotEmpty(t, v, "all values should be populated")
+	}
```

Comment:
Yes, that's the plan.  Completely robust testing (ie, that exercises the use of these maps to detect collisions) will need to wait until we handle in-place restores, since we still can't restore to existing containers.  Until that point you'll see primarily smoke and mock testing in these PRs.

Reply:
given that I found two instances where we weren't fetching enough data to actually populate the unique key as expected I'd say that these should be non-smoke tests. It seems too easy to miss and error-prone to catch in higher layers

Reply:
That's fair, I can bulk them up a bit.